rust/src/circuitbreaker.rs

```diff
@@ -0,0 +1,103 @@
+use std::time::{Duration, Instant};
+
+#[derive(Debug, Clone, Copy)]
+pub struct Node {
+	pub expires: Instant,
+	pub error_count: usize,
+	pub total_count: usize,
+}
+
+impl Node {
+	pub fn new() -> Self {
+		Self {
+			expires: Instant::now(),
+			error_count: 0,
+			total_count: 0,
+		}
+	}
+}
+
+#[derive(Debug)]
+pub struct RingBuffer {
+	cursor: usize,
+	nodes: Vec<Node>,
+}
+
+impl RingBuffer {
```

Comment:
Is there a way here to update the data in the active node? eg If I want to set a new value for `error_count` on the current cursor how can I do that?

Reply:
Yes. You'd just do:
```rust
RingBuffer::new(5).next().get_cursor().error_count += 1;
```

But id just add a method on it so that we can keep this field private.

Reply:
I just pushed some changes so now you can actually see how you might add an error count. I decided to also call `reset()` on a `Node` when we call the `next()` method so it's not left up to the implementation but is guaranteed by the program.

The test should illustrate clearly: 
https://github.com/simonschwartz/breaker-box/blob/bb7ad5535268bb2ffff19f153d05aa5b3542ae4a/rust/src/ring_buffer.rs#L142-L175

Reply:
This change also ensures you can never add more than 1 error or success result